feat: Add AssetToken contract for RWA use case

rjsrams opened the pull request:
## Description

This PR introduces AssetToken.sol, a mintable ERC20 token designed for Real World Asset (RWA) use cases.
It includes mint() and redeem() functions with access control via OpenZeppelin's Ownable.

## Folder Structure
- contracts/AssetToken.sol
- scripts/deploy.js
- scripts/interact.js
- test/AssetToken.js

Thaleszh commented:
Looking good! Can you just move the new folder you created to `contracts`? It will keeping this organized

rjsrams commented:
✅ Done!
I’ve moved the contract file into the contracts/ folder and updated the project structure accordingly.
README has also been standardized.
Thank you!

Thaleszh commented on README.md:
```diff
@@ -1,73 +1,106 @@
-# 🧾 Kii Solidity Contracts
+# KiiChain RWA Token — AssetToken.sol
```
You overwrote the readme of the whole repo. If you want to keep things modular, you can make a folder `contracts/AssetToken/` and place everything there to include a readme. The important thing is that the repo is not only this

Thaleszh commented on scripts/deploy.js:
```diff
@@ -0,0 +1,26 @@
+require('dotenv').config();
+const { ethers } = require('hardhat');
```
In the same vein, these two deploy and interact scripts are specific to the AssetToken. Either rename them to be more specific or put them into a seperate folder that indicates the context (i.e `/scripts/AssetToken`)

Thaleszh commented on test/AssetToken.test.js:
```diff
@@ -0,0 +1,33 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
```
Test is looking good, no changes required here

rjsrams commented:
Done: moved all AssetToken-related files into their own folder, updated root README, removed duplicates.

rjsrams commented:
README updated with clearer sections and AssetToken details. Feel free to give feedback
